packages/core/rsc/streamable.ui.test.tsx

```diff
@@ -157,3 +177,151 @@ describe('rsc - render()', () => {
     expect(rendered).toMatchSnapshot();
   });
 });
+
+describe('rsc - createStreamableUI()', () => {
+  it('should emit React Nodes that can be updated', async () => {
+    const ui = createStreamableUI(<div>1</div>);
+    ui.update(<div>2</div>);
+    ui.update(<div>3</div>);
+    ui.done();
+
+    const final = getFinalValueFromResolved(
+      await simulateFlightServerRender(ui.value),
+    );
+    expect(final).toMatchInlineSnapshot(`
+      <div>
+        3
+      </div>
+    `);
+  });
+
+  it('should emit React Nodes that can be updated with .done()', async () => {
+    const ui = createStreamableUI(<div>1</div>);
+    ui.update(<div>2</div>);
+    ui.update(<div>3</div>);
+    await sleep(100);
```

Comment:
can the await be on something else, like the ui.update call?

Reply:
`ui.update` calls are always synchronous, but I hear you!